crates/external_services/src/grpc_client.rs

```diff
@@ -76,3 +78,34 @@ impl GrpcClientSettings {
         })
     }
 }
+
+/// Contains grpc headers
+#[derive(Debug)]
+pub struct GrpcHeaders {
+    /// Tenant id
+    pub tenant_id: String,
+    /// Request id
+    pub request_id: Option<String>,
+}
+
+impl GrpcHeaders {
+    /// Method to add necessary headers to the tonic Request
+    pub fn add_headers_to_grpc_request<T>(&self, request: &mut tonic::Request<T>) {
+        self
+            .tenant_id
+            .parse()
+            .map(|tenant_id| request.metadata_mut().append("x-tenant-id", tenant_id))
+            .inspect_err(|err| logger::warn!(header_parse_error=?err,"invalid x-tenant-id received in update_elimination_bucket_config"))
```

Comment:
Nit: Update the error messages in both cases, remove the `update_elimination_bucket_config` part.

I guess you can also add `#[track_caller]` attribute on this method?